### PR TITLE
Validate function params

### LINKED ISSUE #44: Validate all lib method call parameters
We could bring a package like [this](https://www.npmjs.com/package/node-validator) in and extend it to use the web3 utils checkers like isAddress. Ideally it should have a minimalist presence at the top of every method, like a single call that takes an object with type description mapped to variable and returns a Boolean or something.

### PR DESCRIPTION
**Pull request by MatiasOS**
This resolve #44 
Add [Joi](https://github.com/hapijs/joi) as dependency and update Chai from 3.5.0 to 4.1.2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,9 +36,11 @@
     "babel-polyfill": "^6.23.0",
     "babel-preset-es2015": "^6.3.13",
     "babel-register": "^6.26.0",
-    "chai": "^3.5.0",
+    "chai": "^4.1.2 ",
     "chai-string": "^1.4.0",
     "currency-codes": "^1.1.2",
+    "joi": "^13.1.2",
```
Comment (AugustoL) [inline package.json]: what is joi ??? 0.o why are we using it @MatiasOS ?

**Comment by JirkaChadima** [comment]
@MatiasOS There's a lot of duplicate code in validators... It is a good idea to have exposed named functions, but the regexp's should not be copy-pasted all over the place. Validations for particular types should be in one place and the named validator functions should only reference them. That will bring us a more readable code. But the usage in the library itself looks ok to me.

File: package.json (hunk 1 above)
Comment (MatiasOS) [reply package.json]: Joi is a library that allows us to create schemes for validations. Has excellent functionality to check types and domain restrictions.

**Comment by JirkaChadima** [comment]
@MatiasOS That looks better! And still can be improved :) Notice that for example `hotelAddress` and `unitAddress` are used together a lot, so there's a place for one more abstraction level, I think. You could create something like:

```javascript
const unitAddressValidator = {
    hotelAddress: addressValidator,
    unitAddress: addressValidator,
}

...
// and then one of these - it will create the resulting validator with three attributes, if there's a key with the same name, the last one is used in the result
const unitCostValidator = _.merge(unitAddressValidator, {price: monetaryValidator});
const unitCostValidator = Object.assign({}, unitAddressValidator, {price: monetaryValidator}); // not sure if we can use this in our JS version
```

It will make changing things and adding new validators much easier, you'll just have to combine some existing ones.

**Comment by JirkaChadima** [comment]
Looks good to me... @AugustoL, if you are ok with it as well, please accept and merge this. Good job, @MatiasOS!

**Pull request merged**